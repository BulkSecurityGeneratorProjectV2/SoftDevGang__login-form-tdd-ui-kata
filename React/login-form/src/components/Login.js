import React from 'react';

export const cleaner = (dirty) => {
    const cleaned = dirty;

    return cleaned;
};

export class Login extends React.Component {

    validateInput() {
        this.props.onValidate('userName', 'password');
        console.log("I was called")
    }

    render() {
        return (
            <div>
                <input type={"text"}
                       name={"userName"}
                       placeholder={"user name"}
                       required={"required"}>
                </input>
                <label htmlFor={"userName"}>
                    User name:
                </label>

                <input type={"text"}
                       name={"password"}
                       placeholder={"password"}
                       required={"required"}>
                </input>
                <label htmlFor={"password"}>
                    Password:
                </label>

                <input type={"button"}
                       name={"submit"}
                       value={"Sign in"}
                       onClick={() => this.validateInput()}>
                </input>
            </div>
        );
    }
}


const DefaultFixME = props => <Login onValidate={() => console.log("Fix me")} />;

export default DefaultFixME