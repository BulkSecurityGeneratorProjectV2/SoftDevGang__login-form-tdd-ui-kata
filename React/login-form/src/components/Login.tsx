import React from "react";
import {AuthenticateResult} from "./Authenticate";

interface Props {
    failedLogin: boolean,
    authenticate: (userName: string, password: string) => AuthenticateResult;
}

interface State {
    userName: string;
    password: string;
}

class Login extends React.Component<Props, State> {

    constructor(props: Readonly<Props>) {
        super(props);
        this.state = {
            userName: "",
            password: ""
        };
    }

    public render() {
        const userName = this.state.userName;
        const password = this.state.password;
        const failedLogin = this.props.failedLogin;

        let errorMessage: React.ReactFragment;
        if (failedLogin) {
            errorMessage = <div aria-label={"Failed login"}>
                You are not logged in
            </div>
        } else {
            errorMessage = <div/>
        }

        return <div>
            <label htmlFor={"userName"}>Phone, email or username</label>
            <input id={"userName"}
                   type={"text"}
                   value={userName}
                   onChange={(event) => this.setUserName(event)}
            />
            <label htmlFor={"password"}>Password</label>
            <input id={"password"}
                   type={"password"}
                   value={password}
                   onChange={(event) => this.setPassword(event)}
            />

            {errorMessage}

            <button onClick={() => this.authenticate()}>
                Login
            </button>

        </div>;
    }

    private authenticate(): void {
        const userName: string = this.state.userName;
        const password: string = this.state.password;

        this.props.authenticate(userName, password);
    }

    private setUserName(event: React.ChangeEvent<HTMLInputElement>) {
        const userName: string = event.currentTarget.value;
        this.setState({userName});
    }

    private setPassword(event: React.ChangeEvent<HTMLInputElement>) {
        const password: string = event.currentTarget.value;
        this.setState({password: password});
    }
}

export default Login;
