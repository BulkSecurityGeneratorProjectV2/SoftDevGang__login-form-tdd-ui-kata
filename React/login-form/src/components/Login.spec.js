import React from 'react';
import {cleaner, Login} from './Login';
import renderer from 'react-test-renderer';

import {fireEvent, render} from '@testing-library/react'

describe('Login', () => {


    describe('Rendering', () => {
        test('login should render properly', () => {

            const component = renderer.create(
                <Login/>,
            );
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();

        });

    });

    describe('Submitting', () => {
        test('Field should not be empty', () => {

            const validateFn = jest.fn();

            const {getByText} = render(<Login onValidate={validateFn}/>);
            fireEvent.click(getByText('Sign in'));

            expect(validateFn).toHaveBeenCalled();


        });

    });

    test('Clean a string', () => {
        expect(cleaner("foo")).toEqual("foo");
    });

    test('login should', () => {
        /*
                const component = renderer.create(
                    <Link page="http://www.facebook.com">Facebook</Link>,
                );
                let tree = component.toJSON();
                expect(tree).toMatchSnapshot();

                // manually trigger the callback
                tree.props.onMouseEnter();
                // re-rendering
                tree = component.toJSON();
                expect(tree).toMatchSnapshot();

                // manually trigger the callback
                tree.props.onMouseLeave();
                // re-rendering
                tree = component.toJSON();
                expect(tree).toMatchSnapshot();

         */
    });


});


