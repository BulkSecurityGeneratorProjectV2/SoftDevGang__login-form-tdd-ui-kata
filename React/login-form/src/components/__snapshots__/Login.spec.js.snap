// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Rendering login should render properly 1`] = `
<div>
  <input
    name="userName"
    placeholder="user name"
    required="required"
    type="text"
  />
  <label
    htmlFor="userName"
  >
    User name:
  </label>
  <input
    name="password"
    placeholder="password"
    required="required"
    type="text"
  />
  <label
    htmlFor="password"
  >
    Password:
  </label>
  <input
    name="submit"
    onClick={[Function]}
    type="button"
    value="Sign in"
  />
</div>
`;
